alf/algorithms/sarsa_algorithm.py

```diff
@@ -440,11 +440,13 @@ def calc_loss(self, experience, info: SarsaInfo):
 
         critic_loss = math_ops.add_n(critic_losses)
 
-        not_first_step = (experience.step_type != StepType.FIRST)
-        critic_loss = critic_loss * not_first_step.to(torch.float32)
+        not_first_step = (experience.step_type != StepType.FIRST).to(
+            torch.float32)
+        critic_loss = critic_loss * not_first_step
         if (experience.batch_info != ()
                 and experience.batch_info.importance_weights != ()):
-            priority = critic_loss.sum(dim=0).sqrt()
+            valid_n = torch.clamp(not_first_step.sum(dim=0), min=1.0)
+            priority = (critic_loss.sum(dim=0) / valid_n).sqrt()
```

Comment:
There is still some issue. For mini_batch_length=1, priority will always be 0 if step_type is FIRST_STEP. Hence this kind of samples will not be sampled anymore. Though this does not affect the training of critic, it may affect the training of actor. Since the actor loss will not contain samples at the FIRST_STEP anymore. I am not sure what is best way to fix this.